notification-rule-config-service-api/src/main/proto/org/hypertrace/notification/config/service/v1/notification_rule.proto

```diff
@@ -16,6 +16,25 @@ message NotificationRuleMutableData {
   string event_condition_id = 3;
   string event_condition_type = 4;
 
-  string channel_id = 5;
+  string channel_id = 5 [deprecated=true];
   string rate_limit_interval_duration = 6;
+
+  oneof notification_target {
```

Comment:
@aaron-steinfeld Based on the slack discussion (pasted the snippet below), hope we can go forward with this?

> now one particular notification condition requires a special notification config

that’s how the requirement goes..

> Are they actually agnostic though? every notification type can be sent to every channel already?

yes AFAIK and from the user UI perspective too

> Agree an enum is better here, but since both would be placeholders until we migrate to policy engine, it doesn’t seem like it’s a significant amount of tech debt to take on. The whole idea of a channel is the real tech debt.

if that’s anyways true, there shouldn’t be any harm to have an enum.. that would be clearer of the 2 options..

Reply:
> if that’s anyways true, there shouldn’t be any harm to have an enum.. that would be clearer of the 2 options..
Which Enum are we referring to?

Reply:
> Which Enum are we referring to?

The integration type - my preference was to use a string there so this notification is not tightly coupled with integration support as it's just one more place to change when adding a new type (since the only code that actually needs to know about the type is the code that's evaluating the rule).

Reply:
@aaron-steinfeld currently channel allows co-existence of multiple components - say email+webhook+splunk.. But in this case, Wiz cannot coexist with these other components because of its different behaviour. I was preferring keeping this restriction as part of the API rather than the channel implementation..

Reply:
So an event being pushed to wiz cannot be sent as an email? If so, I think it's fine to leave it in the rule,  but my preference would be something like

```proto
message NotificationRuleMutableData {
  string rule_name = 1;
  string description = 2;

  string event_condition_id = 3;
  string event_condition_type = 4;

  optional string channel_id = 5;
  string rate_limit_interval_duration = 6;
  NotificationIntegrationTarget integration_target = 7;
}

message NotificationIntegrationTarget {
  string type = 1;
  string integration_id = 2;
}

```

Reply:
> You can add the integration target as a sibling and un-deprecate the channel id to dramatically reduce the blast radius of this change.

Here can we leave channel_id as is and introduce channelType, something like:
```
message ChannelType {
  CHANNEL_TYPE_COMBINATION = 0;
  CHANNEL_TYPE_WIZ_INTEGRATION = 1;
}
```
I know it's a bit hacky, but @aaron-steinfeld does this work? then `channel_id` can be reused as integration_id in wiz case..

Reply:
^^^ works for me.. missed the part that we can add optional to string without breaking API.. @DibyojyotiS let's go with this..

Reply:
it's _technically_ a breaking change, but I think the functional impact is less than if we left it deprecated and had a second way of referencing a channel.

Reply:
I made the change - replaced the enum with a string.

Reply:
The second part was unwinding the one of and leaving the channel field as is, just adding the optional qualifier - see above

Reply:
> it's technically a breaking change, but I think the functional impact is less than if we left it deprecated and had a second way of referencing a channel.

Oh, my bad. I had incorrectly read this comment - somehow I missed the word "than" and it changed the meaning. Removed the oneof too.